atlasdb-config/src/test/java/com/palantir/atlasdb/config/ShouldRunBackgroundSweepSupplierTest.java

```diff
@@ -0,0 +1,119 @@
+                SWEEP_QUEUE_WRITES_DISABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_DISABLED, BACKGROUND_SWEEP_DISABLED));
+        assertThat(supplier.get()).isFalse();
+
+        ShouldRunBackgroundSweepSupplier supplier2 = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_ENABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_DISABLED));
+        assertThat(supplier2.get()).isFalse();
+    }
+
+    @Test
+    public void enableBackgroundSweepIfBackgroundSweepExplicitlyEnabled() {
+        ShouldRunBackgroundSweepSupplier supplier = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_DISABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_DISABLED, BACKGROUND_SWEEP_ENABLED));
+        assertThat(supplier.get()).isTrue();
+
+        ShouldRunBackgroundSweepSupplier supplier2 = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_ENABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_ENABLED));
+        assertThat(supplier2.get()).isTrue();
+    }
+
+    @Test
+    public void enableBackgroundSweepIfNotSetAndTargetedSweepNotFullyEnabled() {
+        ShouldRunBackgroundSweepSupplier supplier = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_ENABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_DISABLED, BACKGROUND_SWEEP_UNSET));
+        assertThat(supplier.get()).isTrue();
+
+        ShouldRunBackgroundSweepSupplier supplier2 = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_DISABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_UNSET));
+        assertThat(supplier2.get()).isTrue();
+    }
+
+    @Test
+    public void disableBackgroundSweepIfNotSetAndTargetedSweepFullyEnabled() {
+        ShouldRunBackgroundSweepSupplier supplier = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_ENABLED,
+                () -> createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_UNSET));
+        assertThat(supplier.get()).isFalse();
+    }
+
+    @SuppressWarnings("unchecked") // Mock assignment known to be safe
+    @Test
+    public void liveReload() {
+        Supplier<AtlasDbRuntimeConfig> runtimeConfigSupplier = mock(Supplier.class);
+        when(runtimeConfigSupplier.get())
+                .thenReturn(createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_UNSET))
+                .thenReturn(createRuntimeConfig(TARGETED_SWEEP_DISABLED, BACKGROUND_SWEEP_UNSET))
+                .thenReturn(createRuntimeConfig(TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_ENABLED));
+
+        ShouldRunBackgroundSweepSupplier supplier = new ShouldRunBackgroundSweepSupplier(
+                SWEEP_QUEUE_WRITES_ENABLED, runtimeConfigSupplier);
+
+        assertThat(supplier.get()).isFalse();
```

Comment:
nit, add `as` for each assert please, can be as simple as 👍 

```java
assertThat(supplier.get())
  .as("TARGETED_SWEEP_ENABLED, BACKGROUND_SWEEP_UNSET")
  .isFalse();
```

etc.

Reply:
Ah, okay. Sure